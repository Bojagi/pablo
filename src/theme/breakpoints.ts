export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl';

export const breakpointNames: Breakpoint[] = ['sm', 'md', 'lg', 'xl'];

export type BreakpointsArray = [number, number, number, number];

export type Breakpoints = BreakpointsArray &
  Record<Breakpoint, number> & {
    breakpointNames: Breakpoint[];
  };

export function createBreakpoints(
  array: BreakpointsArray,
  usedBreakpointNames: Breakpoint[] = breakpointNames
): Breakpoints {
  const newBreakpoints = ([...array] as unknown) as Breakpoints;
  usedBreakpointNames.forEach((bp, index) => {
    newBreakpoints[bp] = newBreakpoints[index];
  });
  newBreakpoints.breakpointNames = usedBreakpointNames;

  return newBreakpoints as Breakpoints;
}

export const defaultBreakpoints: BreakpointsArray = [600, 1000, 1200, 1920];

export const breakpoints = createBreakpoints(defaultBreakpoints, breakpointNames);
