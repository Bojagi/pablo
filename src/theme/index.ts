export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
