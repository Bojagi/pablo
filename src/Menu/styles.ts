import { registerComponentStyles } from '../theme/defaultComponentStyles';
import { BaseStyles } from '../types';

export type MenuItemStyleProperties = 'item' | 'itemText';
export type MenuStyleProperties = 'box' | 'arrow' | MenuItemStyleProperties;

export interface MenuStyles extends BaseStyles<MenuStyleProperties> {}

export const menuStyles: MenuStyles = {};

registerComponentStyles('menu', menuStyles);
