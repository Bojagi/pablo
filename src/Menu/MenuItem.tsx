import React, { ReactNode } from 'react';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { buttonBaseStyles } from '../ButtonBase';
import { getSpacing } from '../styleHelpers';
import { themeVars } from '../theme/themeVars';
import { BaseProps } from '../types';
import { ButtonTypography } from '../Typography';
import { getCustomStyles, useCustomStyles } from '../utils/useCustomStyles';
import { MenuItemStyleProperties } from './styles';

export interface MenuItemProps extends BaseProps<MenuItemStyleProperties> {
  children: ReactNode;
  onClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
  selected?: boolean;
  as?: React.ElementType;
}

const MenuItemBox = styled.div<MenuItemProps>`
  ${buttonBaseStyles as any}
  display: block;
  color: ${themeVars.colors.common.black};
  text-decoration: none;
  margin-top: ${getSpacing(0.25)};
  margin-bottom: ${getSpacing(0.25)};
  padding: ${getSpacing(0.5)};

  ${(props: MenuItemProps) =>
    props.selected
      ? css`
          background-color: ${themeVars.colors.brand.lightest};
        `
      : css`
          &:hover {
            background-color: ${themeVars.colors.gray['50']};
          }

          &:active {
            background-color: ${themeVars.colors.gray[100]};
          }
        `}
  ${getCustomStyles('menu.styles', 'item')}
`;

export const MenuItem = ({ children, onClick, as, customStyles, ...props }: MenuItemProps) => (
  <MenuItemBox
    data-testid="pbl-menu-item"
    as={as}
    role="menuitem"
    onClick={onClick}
    customStyles={customStyles}
    {...props}
  >
    <ButtonTypography
      customStyles={{
        button: useCustomStyles('menu.styles', customStyles)('itemText'),
      }}
    >
      {children}
    </ButtonTypography>
  </MenuItemBox>
);
