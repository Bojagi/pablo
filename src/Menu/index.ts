import './styles';

export * from './Menu';
export * from './MenuItem';
