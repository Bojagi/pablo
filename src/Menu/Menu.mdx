---
name: Menu
route: /menu
menu: Components
---

import { Props, Playground } from 'docz'
import { Box } from '../Box/Box.tsx'
import { Button } from '../Button/Button.tsx'
import { Menu } from './Menu.tsx'
import { MenuItem } from './MenuItem.tsx'
import { FadeAnimation, SlideAnimation } from '../animation/index.tsx'
import { Radio, RadioGroup } from '../Radio/index.ts'
import { Home, ShoppingCart, BookOpen } from 'react-feather';
import { css } from '@emotion/react';

# Menu

Menus are used for adding a list of choices to an element that is visible on click.  It should be used for
context specific actions.

## Examples

### Simple Menu

A menu needs to be attached to an element. This is done by placing the element as a child, 
it will automatically take the ref. This will change in the future though in favor of passing a ref
and using the items as children. For now `MenuItem` elements are passed as an array in the `items` prop.

<Playground>
  {() => {
    const [open, setOpen] = React.useState(false);
    const handleClose = () => setOpen(false);
    return (
      <Menu
      open={open}
      offset={-5}
      onClose={handleClose}
      items={[
        <MenuItem key="1" onClick={handleClose}>
          User Profile
        </MenuItem>,
        <MenuItem key="2" onClick={handleClose}>
          Settings
        </MenuItem>,
        <MenuItem key="3" onClick={handleClose}>
          Logout
        </MenuItem>,
      ]}
    >
      <Button onClick={() => setOpen(!open)}>Click me</Button>
    </Menu>
    );
  }}
</Playground>

### Menu placement

Like every Popover style overlay, a `Menu` can be placed on different sides of it's attached element.
For this you use the `placement` prop.

<Playground>
  {() => {
    const [open, setOpen] = React.useState(false);
    const [placement, setPlacement] = React.useState('top');
    const handleClose = () => setOpen(false);
    return (
      <>
        <Menu
          open={open}
          offset={-5}
          onClose={handleClose}
          items={[
            <MenuItem key="1" onClick={handleClose}>
              User Profile
            </MenuItem>,
            <MenuItem key="2" onClick={handleClose}>
              Settings
            </MenuItem>,
            <MenuItem key="3" onClick={handleClose}>
              Logout
            </MenuItem>,
          ]}
          placement={placement}
        >
          <Button onClick={() => setOpen(!open)}>Please click on me</Button>
        </Menu>
        <RadioGroup size="small" my={4} value={placement} onChange={setPlacement}>
          <Radio value="top" label="top" />
          <Radio value="top-start" label="top-start" />
          <Radio value="top-end" label="top-end" />
          <Radio value="right" label="right" />
          <Radio value="right-start" label="right-start" />
          <Radio value="right-end" label="right-end" />
          <Radio value="bottom" label="bottom" />
          <Radio value="bottom-start" label="bottom-start" />
          <Radio value="bottom-end" label="bottom-end" />
          <Radio value="left" label="left" />
          <Radio value="left-start" label="left-start" />
          <Radio value="left-end" label="left-end" />
        </RadioGroup>
      </>
    );
  }}
</Playground>

### Menu with animation

You can easily add animations to your menu with the Pablo animation components. Pass a Pablo animation
component like `SlideAnimation` or `FadeAnimation` to the `animation` prop and with the `animationProps`
prop you configure animation properties such as `duration` and `side` of the animation.

<Playground>
  {() => {
    const [open, setOpen] = React.useState(false);
    const handleClose = () => setOpen(false);
    return (
      <Menu
      open={open}
      offset={-5}
      animation={SlideAnimation}
      animationProps={{
        duration: 150,
        side: 'bottom',
      }}
      onClose={handleClose}
      items={[
        <MenuItem key="1" onClick={handleClose}>
          User Profile
        </MenuItem>,
        <MenuItem key="2" onClick={handleClose}>
          Settings
        </MenuItem>,
        <MenuItem key="3" onClick={handleClose}>
          Logout
        </MenuItem>,
      ]}
    >
      <Button onClick={() => setOpen(!open)}>Click me</Button>
    </Menu>
    );
  }}
</Playground>

## Import

```js
import { Menu, MenuItem } from '@bojagi/pablo/Menu';
```

## Props

### Menu
<Props of={Menu} />

### MenuItem
<Props of={MenuItem} />
