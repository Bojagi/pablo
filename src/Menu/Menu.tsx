import { Placement } from '@popperjs/core';
import React, { forwardRef, ReactComponentElement, ReactElement } from 'react';

import styled from 'styled-components';
import { Box } from '../Box';
import { Popover, PopoverProps } from '../Popover/Popover';
import { PopoverArrow } from '../PopoverArrow/PopoverArrow';
import { getColor } from '../styleHelpers';
import { BaseProps } from '../types';
import { useCustomStyles } from '../utils/useCustomStyles';
import { MenuItemProps } from './MenuItem';
import { MenuStyleProperties } from './styles';

export interface MenuProps extends BaseProps<MenuStyleProperties> {
  children: ReactComponentElement<any>;
  open: boolean;
  placement: Placement;
  offset?: number;
  onClose?: () => void;
  items: ReactElement<MenuItemProps>[];
  animation?: PopoverProps['animation'];
  animationProps?: PopoverProps['animationProps'];
}

const MenuBox = styled(Box)`
  background-color: ${getColor('common', 'white')};
  border-radius: 4px;
  max-width: 400px;
  min-width: 120px;
  padding: 8px;
  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.2);
  ${(props) => props.css}
`;

export const Menu = forwardRef(
  (
    {
      children,
      items,
      open,
      offset,
      onClose,
      placement,
      animation,
      animationProps,
      customStyles,
    }: MenuProps,
    ref
  ) => {
    const getCustomStyles = useCustomStyles('menu.styles', customStyles);
    return (
      <Popover
        ref={ref}
        open={open}
        offset={offset}
        onClickOutside={onClose}
        arrow={<PopoverArrow size={10} color="white" customCss={getCustomStyles('arrow')} />}
        placement={placement}
        content={<MenuBox css={getCustomStyles('box')}>{items}</MenuBox>}
        animation={animation}
        animationProps={animationProps}
      >
        {children}
      </Popover>
    );
  }
);
