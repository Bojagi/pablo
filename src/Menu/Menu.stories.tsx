import React, { useCallback, useState } from 'react';
import { css } from '@emotion/react';
import { SlideAnimation } from '../animation/SlideAnimation';
import { Button } from '../Button';
import { Menu } from './Menu';
import { MenuItem } from './MenuItem';

export default {
  title: 'Menu',
  parameters: {
    layout: 'centered',
  },
};

const BaseStory = ({ menuItemProps, ...args }) => {
  const [open, setOpen] = useState(false);

  const handleClose = useCallback(() => setOpen(false), []);

  return (
    <Menu
      open={open}
      offset={-5}
      onClose={handleClose}
      items={[
        <MenuItem key="1" onClick={handleClose} {...menuItemProps}>
          User Profile
        </MenuItem>,
        <MenuItem key="2" onClick={handleClose} {...menuItemProps}>
          Settings
        </MenuItem>,
        <MenuItem key="3" onClick={handleClose} {...menuItemProps}>
          Logout
        </MenuItem>,
      ]}
      placement="top"
      {...args}
    >
      <Button onClick={() => setOpen(!open)}>Click me</Button>
    </Menu>
  );
};

export const BottomEnd = BaseStory.bind(null);
BottomEnd.args = {
  placement: 'bottom-end',
};

export const BottomStart = BaseStory.bind(null);
BottomStart.args = {
  placement: 'bottom-start',
};

export const Bottom = BaseStory.bind(null);
Bottom.args = {
  placement: 'bottom',
};

export const TopEnd = BaseStory.bind(null);
TopEnd.args = {
  placement: 'top-end',
};

export const TopStart = BaseStory.bind(null);
TopStart.args = {
  placement: 'top-start',
};

export const Top = BaseStory.bind(null);
Top.args = {
  placement: 'top',
};

export const RightStart = BaseStory.bind(null);
RightStart.args = {
  placement: 'right-start',
};

export const RightEnd = BaseStory.bind(null);
RightEnd.args = {
  placement: 'right-end',
};

export const Right = BaseStory.bind(null);
Right.args = {
  placement: 'right',
};

export const Left = BaseStory.bind(null);
Left.args = {
  placement: 'left',
};

export const LeftStart = BaseStory.bind(null);
LeftStart.args = {
  placement: 'left-start',
};

export const LeftEnd = BaseStory.bind(null);
LeftEnd.args = {
  placement: 'left-end',
};

export const WithAnimation = BaseStory.bind(null);
WithAnimation.args = {
  placement: 'bottom-end',
  animation: SlideAnimation,
  animationProps: {
    duration: 150,
    side: 'bottom',
  },
};

export const WithCustomStyles = BaseStory.bind(null);
WithCustomStyles.args = {
  placement: 'bottom-end',
  animation: SlideAnimation,
  animationProps: {
    duration: 150,
    side: 'bottom',
  },
  menuItemProps: {
    customStyles: {
      item: css`
        background-color: blue;
      `,
      itemText: css`
        color: yellow;
      `,
    },
  },
  customStyles: {
    box: css`
      background-color: red;
    `,
    arrow: css`
      border-bottom-color: red;
    `,
  },
};
