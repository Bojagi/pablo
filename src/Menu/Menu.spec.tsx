import {
  render,
  getByTestId as globalGetByTestId,
  getAllByTestId as globalGetAllByTestId,
  act,
  fireEvent,
} from '@testing-library/react';
import React from 'react';
import { PabloThemeProvider } from '../theme';
import { Menu } from './Menu';
import { MenuItem } from './MenuItem';
import '../../testUtils/mockResizeObserver';
import { waitOneTick } from '../../testUtils/waitOneTick';

let menuItemClickHandler;

beforeEach(() => {
  menuItemClickHandler = jest.fn();
});

test('Render closed Menu', () => {
  const { container, getByTestId } = renderComponent({});
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-popover-mountpoint')).toBeEmptyDOMElement();
});

test('Render open Menu', async () => {
  const { container, getByTestId } = renderComponent({ open: true });
  await waitOneTick();
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-popover-mountpoint')).not.toBeEmptyDOMElement();
});

test('Click on menu item', async () => {
  const { container, getAllByTestId } = renderComponent({ open: true });
  await waitOneTick();
  expect(container).toMatchSnapshot();
  act(() => {
    fireEvent.click(getAllByTestId('pbl-menu-item')[0]);
  });
  await waitOneTick();
  expect(menuItemClickHandler).toHaveBeenCalledTimes(1);
  expect(menuItemClickHandler).toHaveBeenCalledWith('hi');
});

function renderComponent(props) {
  const { baseElement, unmount } = render(
    <PabloThemeProvider>
      <Menu
        {...props}
        items={[
          <MenuItem key="1" onClick={() => menuItemClickHandler('hi')}>
            Hello
          </MenuItem>,
          <MenuItem key="2" onClick={() => menuItemClickHandler('earth')}>
            World
          </MenuItem>,
        ]}
      >
        <div>Content</div>
      </Menu>
    </PabloThemeProvider>
  );

  return {
    getByTestId: globalGetByTestId.bind(null, baseElement),
    getAllByTestId: globalGetAllByTestId.bind(null, baseElement),
    unmount,
    container: baseElement,
  };
}
