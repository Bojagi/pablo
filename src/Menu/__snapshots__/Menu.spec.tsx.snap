// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click on menu item 1`] = `
.c2 {
  box-sizing: border-box;
}

.c1 {
  visibility: hidden;
}

.c0 {
  box-sizing: border-box;
  z-index: 1100;
}

.c6 {
  box-sizing: border-box;
  width: 10px;
  height: 10px;
}

.c6:after {
  box-sizing: border-box;
  content: '';
  position: absolute;
  top: -10;
  left: -5px;
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;
  border-bottom: 10px solid white;
}

*[data-popper-placement^='top'] .c6 {
  bottom: -10px;
}

*[data-popper-placement^='top'] .c6:after {
  transform: rotate(180deg);
}

*[data-popper-placement^='bottom'] .c6 {
  top: -10px;
}

*[data-popper-placement^='left'] .c6 {
  right: -10px;
}

*[data-popper-placement^='left'] .c6:after {
  transform: rotate(90deg) scaleX(0.75);
}

*[data-popper-placement^='right'] .c6 {
  left: -10px;
}

*[data-popper-placement^='right'] .c6:after {
  transform: rotate(-90deg) scaleX(0.75);
}

.c3 {
  box-sizing: border-box;
  background-color: var(--pbl-theme-colors-common-white);
  border-radius: 4px;
  max-width: 400px;
  min-width: 120px;
  padding: 8px;
  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.2);
}

.c5 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-style: normal;
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin: 0;
  font-size: var(--pbl-theme-typography-button-fontSize);
  line-height: var(--pbl-theme-typography-button-lineHeight);
  margin-bottom: var(--pbl-theme-typography-button-marginBottom);
}

.c4 {
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  text-decoration: none;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  transition: background-color,0.3s,color 0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out,transform,0.15s;
  outline: none;
  display: block;
  color: var(--pbl-theme-colors-common-black);
  text-decoration: none;
  margin-top: 2px;
  margin-bottom: 2px;
  padding: 4px;
}

.c4:enabled {
  cursor: pointer;
}

.c4:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c4:hover {
  background-color: var(--pbl-theme-colors-gray-50);
}

.c4:active {
  background-color: var(--pbl-theme-colors-gray-100);
}

<body>
  <div>
    <div
      aria-haspopup="menu"
    >
      Content
    </div>
  </div>
  <div
    data-testid="pbl-popover-mountpoint"
  >
    <div
      class="c0"
      data-popper-escaped="true"
      data-popper-placement="bottom"
      data-popper-reference-hidden="true"
      style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 10px);"
    >
      <div
        basestyles=""
        class="c1"
        data-testid="pbl-animation-inner"
        duration="0"
        enterstyles=""
        exitstyles="
  visibility: hidden;
"
        state="exited"
      >
        <div>
          <div
            class="c2 c3"
            css=","
            role="menu"
          >
            <div
              class="c4"
              data-testid="pbl-menu-item"
              role="menuitem"
            >
              <span
                class="c5"
                customstyles="[object Object]"
              >
                Hello
              </span>
            </div>
            <div
              class="c4"
              data-testid="pbl-menu-item"
              role="menuitem"
            >
              <span
                class="c5"
                customstyles="[object Object]"
              >
                World
              </span>
            </div>
          </div>
        </div>
        <div
          class="c6"
          color="white"
          customcss=","
          size="10"
          style="position: absolute; left: 0px; transform: translate(10px, 0px);"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Render closed Menu 1`] = `
<body>
  <div>
    <div
      aria-haspopup="menu"
    >
      Content
    </div>
  </div>
  <div
    data-testid="pbl-popover-mountpoint"
  />
</body>
`;

exports[`Render open Menu 1`] = `
.c1 {
  box-sizing: border-box;
}

.c0 {
  box-sizing: border-box;
  z-index: 1100;
}

.c5 {
  box-sizing: border-box;
  width: 10px;
  height: 10px;
}

.c5:after {
  box-sizing: border-box;
  content: '';
  position: absolute;
  top: -10;
  left: -5px;
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;
  border-bottom: 10px solid white;
}

*[data-popper-placement^='top'] .c5 {
  bottom: -10px;
}

*[data-popper-placement^='top'] .c5:after {
  transform: rotate(180deg);
}

*[data-popper-placement^='bottom'] .c5 {
  top: -10px;
}

*[data-popper-placement^='left'] .c5 {
  right: -10px;
}

*[data-popper-placement^='left'] .c5:after {
  transform: rotate(90deg) scaleX(0.75);
}

*[data-popper-placement^='right'] .c5 {
  left: -10px;
}

*[data-popper-placement^='right'] .c5:after {
  transform: rotate(-90deg) scaleX(0.75);
}

.c2 {
  box-sizing: border-box;
  background-color: var(--pbl-theme-colors-common-white);
  border-radius: 4px;
  max-width: 400px;
  min-width: 120px;
  padding: 8px;
  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.2);
}

.c4 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-style: normal;
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin: 0;
  font-size: var(--pbl-theme-typography-button-fontSize);
  line-height: var(--pbl-theme-typography-button-lineHeight);
  margin-bottom: var(--pbl-theme-typography-button-marginBottom);
}

.c3 {
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  text-decoration: none;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  transition: background-color,0.3s,color 0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out,transform,0.15s;
  outline: none;
  display: block;
  color: var(--pbl-theme-colors-common-black);
  text-decoration: none;
  margin-top: 2px;
  margin-bottom: 2px;
  padding: 4px;
}

.c3:enabled {
  cursor: pointer;
}

.c3:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c3:hover {
  background-color: var(--pbl-theme-colors-gray-50);
}

.c3:active {
  background-color: var(--pbl-theme-colors-gray-100);
}

<body>
  <div>
    <div
      aria-haspopup="menu"
    >
      Content
    </div>
  </div>
  <div
    data-testid="pbl-popover-mountpoint"
  >
    <div
      class="c0"
      data-popper-escaped="true"
      data-popper-placement="bottom"
      data-popper-reference-hidden="true"
      style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 10px);"
    >
      <div
        basestyles=""
        class=""
        data-testid="pbl-animation-inner"
        duration="0"
        enterstyles=""
        exitstyles="
  visibility: hidden;
"
        state="entering"
      >
        <div>
          <div
            class="c1 c2"
            css=","
            role="menu"
          >
            <div
              class="c3"
              data-testid="pbl-menu-item"
              role="menuitem"
            >
              <span
                class="c4"
                customstyles="[object Object]"
              >
                Hello
              </span>
            </div>
            <div
              class="c3"
              data-testid="pbl-menu-item"
              role="menuitem"
            >
              <span
                class="c4"
                customstyles="[object Object]"
              >
                World
              </span>
            </div>
          </div>
        </div>
        <div
          class="c5"
          color="white"
          customcss=","
          size="10"
          style="position: absolute; left: 0px; transform: translate(10px, 0px);"
        />
      </div>
    </div>
  </div>
</body>
`;
